src/lib/marshal/translation/processor.isomorphic.ts

```diff
@@ -16,13 +16,18 @@ export const formatRef = (
  * Returns a formatted translation file name based on the given translation ref
  * or payload.
  */
-export const formatFileName = (input: string | TranslationData): string => {
+export const formatFileName = (
+  input: string | TranslationData,
+  format: string | undefined,
+): string => {
   const ref =
     typeof input === "string"
       ? input
       : formatRef(input.locale_code, input.namespace);
 
-  return `${ref}.json`;
+  const extension = format === "po" ? "po" : "json";
+
+  return `${ref}.${extension}`;
```

Comment:
```suggestion
  return `${ref}.${format}`;
```
I think we can use the format value directly. Not opposed to having an explicit format -> extension mapping, but in that case i'd prefer to see a switch statement vs a ternary.

Reply:
Yeah, the typing will make that cleaner